cli/azd/pkg/apphost/testdata/TestAspireContainerGeneration-main.bicep.snap

```diff
@@ -9,26 +9,48 @@ param environmentName string
 @description('The location used for all deployed resources')
 param location string
 
+@metadata({azd: {
+  type: 'autoOrManualGeneration'
```

Comment:
Sort of a nit, but I don't love using `autoOrManualGeneration` as a `type` here, since this doesn't really impact the type of the value that gets pushed in.  In the few other places we use `type` today we do things like `inputs` (for a special type of object value, which you are removing) or `location` (for a string which should be an azure location, that feels more like a type constraint than this).

Part of me wonders if we should do something like:

```
azd: {
    generate: {
       config: {
          length: 10
       }
    }
}
```

If we do want to keep this as a type, perhaps just `generated`?

I'm thinking that perhaps we want to decouple the "you can manually provide a value" part of this out and just say: "You can provide a value for any bicep parameter via some yet to be invented gesture (you could image an `azd infra config` at some point, or something like `azd config --env set params.my_sql_abstract_pas_sw_ord "foo"` which sets `params.my_sql_abstract_pas_sw_ord` in the environment config.json."

This allows configuring any bicep value (even one with a default, as `generate` provides here).

Reply:
The use-case for the auto generated or manual input came from the .NetAspire case where your project might become a github-template. In that case, the appHost won't set a static user-name for a Server, for example.
And, instead of adding a manual step for customers to override the auto-generated value, they can set the value as part of `azd up`.

Note, we have the `NoManual` field for the metadata to force the generation to be the only way. We could invert the logic to make it `off` by default and requiere `.Manuel = true` to allow asking customer for manual input.

The manual input is good b/c it does not make customer to go and find main.parameters.json or the ENV VAR to use for setting the value in advance.

About using the `type`, I can remove it, but, to my mind, the `type` is the shape of the metadata you are setting for the azd-known-filed.  If we don't have it here, what would it happen for `Type=location + generate{ .... } ` ??  Would `generate` be just ignored?   Then when does `generate` actually works?  Only for when it is the only field in metadata?
Having `Type` is an easy way to know and validate the metadata for the field.

Reply:
> We could invert the logic to make it off by default and requiere .Manuel = true to allow asking customer for manual input.

This is what I was asking for in https://github.com/dotnet/aspire/pull/3090#discussion_r1535648486. I want the default case to be like it is today: these "auto generated values" don't get prompts during `azd up`. If a developer does want the person deploying the app to be prompted, they opt-in to it when they say `builder.AddParameter`. But by default, if there is a `default.generate` value in the aspire-manifest.json, then `azd` doesn't prompt for the value. It just generates one.

Reply:
I'd do that :).
Thanks @eerhardt , I was just about to ping you about this!!